### ISSUE TITLE
Incorrect (no-unused-variable)

### ISSUE THREAD

**Issue by rogierschouten** [palantir/tslint#237]
With tslint 1.0.0 the file below yields an incorrect error while the previous version did not. 
I wanted to reproduce this with a smaller file but I was stopped by another but (issue #236).

(no-unused-variable) lib\server\supervisor\lib\create-config.ts[10, 8]: unused variable: 'addressInfo'

```javascript
/**
 * Copyright(c) 2014 Spirit IT BV
 */

/// <reference path="../../../../typings/server_lib.d.ts"/>
"use strict";

import tc = require("timezonecomplete");

import addressInfo = require("../../../common/net/address-info");
import AddressInfo = addressInfo.AddressInfo;
import ProtocolType = addressInfo.ProtocolType;

import redundantConnectionInfo = require("../../../common/net/redundant-connection-info");
import RedundantConnectionInfo = redundantConnectionInfo.RedundantConnectionInfo;

import leaderElectionConfig = require("../../../common/supervisor/leader-election-config");
import leconf = leaderElectionConfig;

import supervisorConfig = require("../../../common/supervisor/supervisor-config");
import sc = supervisorConfig;



/**
 * Leader election configuration
 */
function createLeaderElectorConfiguration(
	serverCount: number,
	timeout: tc.Duration,
	retry: tc.Duration,
	startupDelay: tc.Duration,
	controlTimeout: tc.Duration
	): leconf.LeConfiguration {
	var config: leconf.LeConfiguration = new leconf.LeConfiguration();
	config.controlTimeout = controlTimeout.clone();
	config.startupDelay = (startupDelay ? startupDelay.clone() : null);
	for (var i = 0; i < serverCount; ++i) {
		var otherServers: AddressInfo[][] = [];
		var timeouts: tc.Duration[] = [];
		var retryIntervals: tc.Duration[] = [];
		for (var j = 0; j < serverCount; ++j) {
			if (j !== i) {
				otherServers.push([{ host: "127.0.0.1", port: 4000 + j, type: ProtocolType.udp4 }]);
			} else {
				otherServers.push([]);
			}
			timeouts.push(timeout.clone());
			retryIntervals.push(retry.clone());
		}
		config.servers.push(new leconf.ServerInfo(
			i,
			[{ host: "127.0.0.1", port: 4000 + i, type: ProtocolType.udp4 }],
			otherServers,
			timeouts,
			retryIntervals
			));
	}
	return config;
}

/**
 * Create an example supervisor configuration that runs on one PC.
 *
 * @param serverCount	Number of redundant servers
 * @param proxyCount	Number of proxies per server
 * @param processCount	Number of processes per proxy
 * @param timeout	Network timeout
 * @param retry	Network retry interval
 * @param startupDelay	Leader election start delay
 * @param controlTimeout	Maximum time for all processes to react to leader elector command
 */
export function createSupervisorConfiguration(
	serverCount: number,
	proxyCount: number,
	processCount: number,
	timeout: tc.Duration,
	retry: tc.Duration,
	startupDelay: tc.Duration,
	controlTimeout: tc.Duration
	): sc.SupervisorConfig {
	var i: number;
	var j: number;
	var k: number;

	var config: sc.SupervisorConfig = new sc.SupervisorConfig();

	// general
	config.name = "Poller";
	config.startupDelay = startupDelay.clone();

	// leader election
	config.leaderElectionConfig = createLeaderElectorConfiguration(serverCount, timeout, retry, startupDelay, controlTimeout);

	// supervisor -> proxy
	config.listenOn = [];
	config.proxyAddress = [];
	for (i = 0; i < serverCount; ++i) {

		// address the supervisor listens on
		config.listenOn.push([{ host: "127.0.0.1", port: 9000 + 10 * proxyCount + i, type: ProtocolType.udp4 }]);

		// address the supervisor uses to reach each proxy
		config.proxyAddress.push([]);
		for (j = 0; j < proxyCount; ++j) {
			config.proxyAddress[i].push(new RedundantConnectionInfo(
				[{ host: "127.0.0.1", port: 9000 + 10 * j + i, type: ProtocolType.udp4 }],
				timeout.clone(),
				retry.clone()
				));
		}
	}

	// proxies
	for (j = 0; j < proxyCount; ++j) {
		var proxyConfig = new sc.ProxyConfig();
		proxyConfig.name = "proxy" + j.toString(10) + "a";

		// supervisor connection
		proxyConfig.supervisorAddress = [];
		for (i = 0; i < serverCount; ++i) {
			proxyConfig.supervisorAddress.push(new RedundantConnectionInfo(
				[{ host: "127.0.0.1", port: 9000 + 10 * proxyCount + i, type: ProtocolType.udp4 }],
				timeout.clone(),
				retry.clone()
				));
			proxyConfig.supervisorListenOn.push([{ host: "127.0.0.1", port: 9000 + 10 * j + i, type: ProtocolType.udp4 }]);
		}

		// local processes
		for (k = 0; k < processCount; ++k) {
			proxyConfig.processes.push(new sc.LocalProcessConfig(
				"example process " + k.toString(10),
				"./examples/example-node-process.js",
				[	"-r",
					"<%= redundantUid %>",
					"-p",
					"<%= proxyUid %>",
					"-n",
					"<%= processUid %>"
				],
				".",
				null,
				tc.Duration.seconds(10)));
		}

		config.proxies.push(proxyConfig);
	}

	// one web server per supervisor
	for (i = 0; i < serverCount; ++i) {
		config.webServer.push(new sc.WebServerConfig({
			host: (i === 0 ? "0.0.0.0" : "127.0.0.1"),
			port: 8000 + i,
			type: ProtocolType.tcp
		}));
	}

	config.startupDelay = (startupDelay ? startupDelay.clone() : null);

	return config;
}

```

tslint.json:

```
{
	  "rules": {
		"ban": [true,
			["_", "extend"],
			["_", "isNull"],
			["_", "isDefined"]
		],
		"class-name": true,
		"comment-format": [true,
			"check-space"
		],
		"curly": true,
		"eofline": true,
		"forin": true,
		"indent": [true, 4],
		"interface-name": false, 
		"jsdoc-format": false, 
		"label-position": true,
		"label-undefined": true,
		"max-line-length": [true, 140],
		"no-arg": true,
		"no-bitwise": true,
		"no-console": [true,
			"debug",
			"info",
			"log",
			"warn",
			"error",			
			"time",
			"timeEnd",
			"trace"
		],
		"no-construct": true,
		"no-debugger": true,
		"no-duplicate-key": true,
		"no-duplicate-variable": true,
		"no-empty": true,
		"no-eval": true,
		"no-string-literal": true,
		"no-trailing-comma": true,
		"no-trailing-whitespace": true,
		"no-unused-expression": true,
		"no-unused-variable": true,
		"no-unreachable": true,
		"no-use-before-declare": true,
		"no-var-requires": true,
		"one-line": [true,
			"check-open-brace",
			"check-catch",
			"check-else",
			"check-whitespace"
		],
		"quotemark": [true, "double"],
		"radix": true,
		"semicolon": true,
		"triple-equals": [true, "allow-null-check"],
		"typedef": [true,
			"callSignature",
			"indexSignature",
			"parameter",
			"propertySignature",
			"memberVariableDeclarator"
		],
		"typedef-whitespace": [true,
			["callSignature", "noSpace"],
			["catchClause", "noSpace"],
			["indexSignature", "space"]
		],
		"use-strict": [true,
			"check-module"
		],
		"variable-name": [true,
			"allow-leading-underscore"
		],
		"whitespace": [true,
			"check-branch",
			"check-decl",
			"check-operator",
			"check-separator",
			"check-type"
		]
	  }

	}
```

**Comment by ashwinr** [comment]
this should be fixed in the upcoming release of TSLint (on monday) that upgrades the compiler to 1.4.
